backends/edif/edif.cc

```diff
@@ -215,7 +265,11 @@ struct EdifBackend : public Backend {
 				if (port_it.second == 1)
 					*f << stringf("          (port %s (direction %s))\n", EDIF_DEF(port_it.first), dir);
 				else
-					*f << stringf("          (port (array %s %d) (direction %s))\n", EDIF_DEF(port_it.first), port_it.second, dir);
+				{
+					int bd[2]={0,port_it.second-1};
+					int b[2]={bd[bounds[0]],bd[bounds[1]]};
```

Comment:
Check if `m = design->module(cell_it.second->type)` and `w = m->wire(port_it.first)` exist. Then use `GetSize(w)`, `w->upto` and `w->start_offset` to calculate `range_left` and `range_right` for `edif_names()`.